### ISSUE TITLE
🐛 [Bug] BUG in conversion of `torch.nn.Softmax` when the output dimensionality is equal to 2

### ISSUE THREAD

**Issue by alibool** [pytorch/TensorRT#2695]
conversion of `torch.nn.Softmax` is wrong when the output dimensionality is equal to 2
```
import mmengine
import torch
from mmaction.apis import init_recognizer
import torch_tensorrt

device = torch.device("cuda:0")
dtype = torch.float32

model_config = dict(
    type='Recognizer3D',
    backbone=dict(
        type='UniFormerV2',
        input_resolution=224,
        patch_size=16,
        width=768,
        layers=12,
        heads=12,
        t_size=8,
        dw_reduction=1.5,
        backbone_drop_path_rate=0.0,
        temporal_downsample=False,
        no_lmhra=True,
        double_lmhra=True,
        return_list=[8, 9, 10, 11],
        n_layers=4,
        n_dim=768,
        n_head=12,
        mlp_factor=4.0,
        drop_path_rate=0.0,
        mlp_dropout=[0.5, 0.5, 0.5, 0.5],
        clip_pretrained=True,
        pretrained='ViT-B/16'),
    cls_head=dict(
        type='UniFormerHead',
        dropout_ratio=0.5,
        num_classes=2,  # change to any number greater than 2 is ok
        in_channels=768,
        average_clips='prob'),
    data_preprocessor=dict(
        type='ActionDataPreprocessor',
        mean=[114.75, 114.75, 114.75],
        std=[57.375, 57.375, 57.375],
        format_shape='NCTHW'))

model = init_recognizer(mmengine.Config({"model": model_config}), None, device=torch.device("cuda:0"))
model.to(dtype)
model.eval()


def forward_resize(x1: torch.Tensor):
    feats, _ = model.extract_feat(x1, test_mode=True)
    cls_scores = model.cls_head(feats)
    prob = model.cls_head.average_clip(cls_scores, num_segs=1)
    return prob


model.forward = forward_resize
x1_ = torch.rand(8, 1, 3, 8, 224, 224).to("cuda:0").to(dtype)

with torch.no_grad():
    print(model(x1_).sum(dim=1))
    trt_gm = torch_tensorrt.compile(model,
                                    inputs=(x1_,),
                                    enabled_precisions={dtype},
                                    device=device,
                                    output_format="graph_module",
                                    require_full_compilation=True)
    print(trt_gm(x1_)[0].sum(dim=1))

```
tested on both A100 and V100, torch 2.2.1, cuda 118, torch-tensorrt 2.2.0 tensorrt 8.6.0 mmcv 2.0.0 mmaction 1.1.0. 
To install the related packages, use
```
pip install packaging==23.0 tensorrt==8.6.0 chardet -i https://mirrors.ustc.edu.cn/pypi/web/simple
pip install torch torchvision torchaudio torch-tensorrt --index-url https://download.pytorch.org/whl/cu118
pip install opencv-contrib-python==4.6.0.66 opencv-python==4.8.1.78 -i https://mirrors.aliyun.com/pypi/simple/
#pip install mmcv==2.1.0 -f https://download.openmmlab.com/mmcv/dist/cu118/torch2.1/index.html
pip install mmcv==2.0.0 -f https://download.openmmlab.com/mmcv/dist/cu118/torch2.0/index.html
pip install mmaction2==1.1.0
```
